remotes/fixup.go

```diff
@@ -51,39 +69,39 @@ func fixupImage(ctx context.Context, baseImage bundle.BaseImage, ref reference.N
 	if err != nil {
 		return bundle.BaseImage{}, err
 	}
+	if err := blobSource.setFromImageReference(imageRef); err != nil {
+		return bundle.BaseImage{}, err
+	}
 
-	// Prepare the copier or the mounter
-	copier, err := newImageCopier(ctx, resolver, sourceFetcher, repoOnly.String())
+	// Prepare the copier
+	copier, err := newImageCopier(ctx, resolver, sourceFetcher, repoOnly.String(), notifyEvent)
 	if err != nil {
 		return bundle.BaseImage{}, err
 	}
-	var handler imageHandler = &copier
-	if reference.Domain(imageRef) == reference.Domain(ref) {
-		mounter, err := newImageMounter(ctx, resolver, copier, sourceRepoOnly.Name(), repoOnly.Name())
-		if err != nil {
-			return bundle.BaseImage{}, err
-		}
-		handler = &mounter
-	}
+	handler := &copier
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
 
-	// Walk the source repository and list all the descriptors
-	accumulator := &descriptorAccumulator{}
-	if err := images.Walk(ctx, images.Handlers(accumulator, images.ChildrenHandler(&imageContentProvider{sourceFetcher})), descriptor); err != nil {
+	wq := newWorkQueue(ctx, 4, 50)
```

Comment:
You should at least extract these constants, so you can customize them in follow-up